internal/provider/cisco/nxos/provider.go

```diff
@@ -1737,6 +1661,87 @@ func (p *Provider) DeleteVRF(ctx context.Context, req *provider.VRFRequest) erro
 	return p.client.Delete(ctx, v)
 }
 
+func (p *Provider) EnsureVTEP(ctx context.Context, req *provider.VTEPRequest) error {
+	if req.ProviderConfig == nil {
+		return errors.New("vtep: provider requires NVE config (ref is nil)")
+	}
+
+	n := new(NVE)
+	n.ID = 1
+	n.AdminSt = AdminStDisabled
+	if req.VTEP.Spec.Enabled {
+		n.AdminSt = AdminStEnabled
+	}
+	n.SourceInterface = req.PrimaryInterface.Spec.Name
+	n.AnycastInterface = req.AnycastInterface.Spec.Name
+
+	if req.VTEP.Spec.MulticastGroup != nil {
+		switch req.VTEP.Spec.MulticastGroup.Type {
+		case v1alpha1.MulticastGroupTypeL2:
+			n.McastGroupL2 = req.VTEP.Spec.MulticastGroup.Prefix.Addr().String()
+		case v1alpha1.MulticastGroupTypeL3:
+			n.McastGroupL3 = req.VTEP.Spec.MulticastGroup.Prefix.Addr().String()
+		}
+	}
+
+	switch req.VTEP.Spec.HostReachability {
+	case v1alpha1.HostReachabilityTypeBGP:
+		n.HostReach = HostReachBGP
+	case v1alpha1.HostReachabilityTypeFloodAndLearn:
+		n.HostReach = HostReachFloodAndLearn
+	default:
+		return fmt.Errorf("vtep: invalid evpn host reachability type %q", req.VTEP.Spec.HostReachability)
+	}
+	n.SuppressARP = req.VTEP.Spec.SuppressARP
+
+	// provider-specific settings
+	vc := new(nxosv1alpha1.VTEPConfig)
+	if err := req.ProviderConfig.Into(vc); err != nil {
+		return fmt.Errorf("vtep: decode provider config: %w", err)
+	}
+
+	if vc.Spec.HoldDownTime > 0 {
+		n.HoldDownTime = vc.Spec.HoldDownTime
+	}
```

Comment:
```suggestion
	n.HoldDownTime = vc.Spec.HoldDownTime
```
With the default of 180 we can omit the if check.